nautilus/include/nautilus/common/traceing.hpp

```
@@ -131,7 +131,7 @@ void traceValueDestruction(value_ref ref);
 
 value_ref traceCast(value_ref state, Type resultType);
 
-std::array<uint8_t, 256>& getVarRefMap();
+std::array<uint8_t, 10240>& getVarRefMap();
```

I think we should make this configurable using a cmake option. Rhetorical default should be 128 or 256. setting high values can have high performance problems and indicates problems.

---

TBH, I would maybe have it as a patch for `NebulaStream` then.

---

hm ok we can also merge it for now. I will then make the array dynamic in a followup PR

---

But I would still argue that we should throw an error if more then 1000 variables are created in one trace. But I will make this configurable using an option.